packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js

```diff
@@ -55,28 +62,28 @@ class LightCurveViewerContainer extends Component {
   }
 
   async requestData () {
+    const { onError } = this.props
     try {
-      this.setState({ loading: asyncStates.loading })
+      // this.setState({ loading: asyncStates.loading })
       const url = this.getSubjectUrl()
       const response = await request.get(url)
-      if (!response.ok) {
-        throw new Error('Invalid response')
-      } else {
+      if (response.ok) {
         // Get the JSON data, or (as a failsafe) parse the JSON data if the
         // response is returned as a string
         return response.body || JSON.parse(response.text)
       }
     } catch (error) {
-      return this.onError(error)
+      onError(error)
     }
   }
 
   async handleSubject () {
+    const { onError } = this.props
     try {
       const rawData = await this.requestData()
-      this.onLoad(rawData)
+      if (rawData) this.onLoad(rawData)
```

Comment:
This fails silently if `rawData` is falsy. Should it message that the request failed?

Reply:
Sorry, just to expand on that, it looks to me like `this.requestData()` returns `undefined` for bad responses or errors. Is that intentional? Previously, that would have thrown a new error here but the code now fails silently.

Reply:
Returning the error makes `rawData` truthy and then I have to check if it is actually the data or not an Error object like I was previously doing. If an error is returned by `this.requestData()` and I don't check that it's not an error object, then `this.onLoad` will attempt to use lodash and set state with that error object and then error. That error will be caught, but it's a lodash error, not the error from the data request. 

This is the bit of error handling and bubbling I found hard with async/await. I'm not sure if this is right, but it's how I got it to work.... returning undefined on error from `this.requestData` made it so I was sure that `onLoad` was only called with what I wanted it to.

Reply:
After catching an error and recovering from it, you’d expect `requestData` to return a `rawData` object. If it helps, think of how the return value from a `catch` block is passed to the next `then` block in a promise chain.